server/rpc.go

```diff
@@ -344,17 +353,36 @@ func authenticateBroadcaster(id string) (*authBroadcasterResponse, error) {
 		return nil, errors.New(string(body))
 	}
 
-	webhookRes := &authBroadcasterResponse{}
+	webhookRes := &discoveryAuthWebhookRes{}
 	if err := json.Unmarshal(body, webhookRes); err != nil {
 		return nil, err
 	}
 
+	addToDiscoveryAuthWebhookCache(addr, webhookRes, authTokenValidPeriod)
```

Comment:
Should we only hit the webhook if there is not already a cached response and return the cached response otherwise?

IIUC the flow will look something like this:

1. B calls GetOrchestrator on O
2. O hits the webhook which returns a price and caches the response which will expire when the auth token associated with this OrchestratorInfo response expires
3. O uses the webhook price instead of calling `orch.PriceInfo()` and it is used as an input into the HMAC construction for the recipientRandHash in the TicketParams sent back to B
4. Whenever O receives a payment, it will be able to confirm that the price included was a price it advertised that has not yet expired (based on the TicketParams expiration period)
5. O also returns an OrchestratorInfo message after each segment. Since we want to avoid hitting the webhook (we want to hit the webhook only during discovery and not during the "hot" transcoding flow) here we want to read from a cache instead.

Reply:
Hmm good question. 

We will (under happy conditions) only store cache for an O if authentication passes, in other words it is not a blacklisted node. 

This response can be an empty object or contain `priceInfo`, both are totally fine, this will indicate that the broadcaster is authenticated. 

From this point on I don't think we will ever call the webhook in the transcoding path, only during discovery. 

So yes we will hit the webhook on every `getOrchestratorInfo` call , even if there already is cache available. But I'm not sure whether that's harmful.  Otherwise I felt like updating the cache would rely too much on it being cleaned by the cleanup interval. 

Looking forward to your thoughts.

Reply:
Ah I was concerned about hitting the webhook whenever a new OrchestratorInfo message is created in ServeSegment, but missed the fact that authenticateBroadcaster is only called for GetOrchestratorInfo calls and not in the orchestratorInfo function which is invoked in ServeSegment for each segment.

The only remaining case where the webhook will be hit during the transcoding flow is if B needs to refresh its auth token and/or ticket params. Hitting the webhook when the auth token is expired makes sense since the B needs to re-authenticate. Hitting the webhook when the ticket params are expired is not necessary, but might be ok to avoid having to distinguish between these two cases on the orchestrator side. The # of webhook calls during the transcoding flow for the expired auth token case should be ok given the 30 minute validity period. The # of webhook calls during the transcoding flow for the expired ticket params case will be higher given the shorter validity period there. Okay for now, but just noting to keep in mind.

Reply:
I'd be okay with checking for existing cache in `authenticateBroadCaster()` and/or lengthening ticket params expiry if that sits better.